crates/ef-testing/src/evm_sequencer/evm_state/v1.rs

```diff
@@ -209,17 +208,8 @@ impl Evm for KakarotSequencer {
             )
             .unwrap();
 
-        //TODO: remove CA - EOA distinction
-        let nonce = if implementation == self.environment.contract_account_class_hash.0 {
-            let key = get_storage_var_address(ACCOUNT_NONCE, &[]);
-            self.state_mut().get_storage_at(starknet_address, key)?
-        } else {
-            // We can't throw an error here, because it could just be an uninitialized account.
-            StarkFelt::from(0_u8)
-        };
-
         Ok(U256::from_be_bytes(
-            Into::<FieldElement>::into(nonce).to_bytes_be(),
+            Into::<FieldElement>::into(implementation).to_bytes_be(),
```

Comment:
this is not correct, you're using the class hash of the account as the account's nonce. in that case, you should be getting the nonce from the `ACCOUNT_NONCE` storage variable

Reply:
Instead of getting implementation from ACCOUNT_IMPLEMENTATION, now im getting it from ACCOUNT_NONCE.